File: SumLists2/SumLists2.cpp
#include<bits/stdc++.h>
using namespace std;
#include "Solution.cpp"


int main() {
    int n;
    cin >> n;
    vector<int> list(n);
    for(int i = 0 ; i < n ; ++i) {
        cin >> list[i];
    }
    ListNode* head1 = ListNode().createList(list);

    int m;
    cin >> m;
    vector<int> list2(m);
    for(int i = 0 ; i < m ; ++i) {
        cin >> list2[i];
    }
    assert(list[0] != 0 && list2[0] != 0);
    ListNode* head2 = ListNode().createList(list2);

    ListNode* result = sumLists2(head1 , head2);

    vector<int> resultList = ListNode().extractList(result);
    for(int i = 0 ; i < resultList.size() ; ++i) {
        cout << resultList[i] << " ";
    }
    cout << "\n";
    return 0;
}

File: SumLists2/SumLists2.java
import java.io.*;
import java.util.*;
import crio.ds.List.ListNode;


class SumLists2 {

  public static void  main (String args []) {
    Scanner sc =  new  Scanner (System.in);

    int n1 = sc.nextInt();
    ArrayList <Integer> list1 = new ArrayList<Integer> ();
    int x;
    for ( int i = 0 ; i < n1; i++) {
      x = sc.nextInt();
      list1.add(x);
    }
    ListNode head1 = ListNode.createList(list1);
    int n2 = sc.nextInt();
    ArrayList <Integer> list2 = new ArrayList<Integer> ();

    for ( int i = 0 ; i < n2; i++) {
      x = sc.nextInt();
      list2.add(x);
    }
    ListNode head2 = ListNode.createList(list2);
    ListNode head = Solution.sumLists1(head1, head2);
    List<Integer> nums = ListNode.extractList(head);

    for ( int i = 0; i < nums.size() ; i++) {
      System.out.print(nums.get(i)  + " ");
    }

  }

}
